src/monattack.cpp

```diff
@@ -425,109 +425,258 @@ void mattack::smash(monster *z, int index)
-            add_msg(m_warning, _("The %s drops a flask of acid!"), z->name().c_str());
+
+        // (1) Give the target a chance at an uncanny_dodge.
+        // (2) If that fails, always fail to dodge 1 in dodge_skill times.
+        // (3) If ok, dodge if dodge_skill > att_rad_dodge_diff.
+        // (4) Otherwise, fail 1 in (att_rad_dodge_diff - dodge_skill) times.
+        if (foe->uncanny_dodge()) {
+            break;
         }
-        z->moves -= 100;
-        for (auto &i : free) {
-            g->m.add_field(i.x, i.y, fd_acid, 3);
+
+        int const  dodge_skill  = foe->get_dodge();
+        bool const critial_fail = one_in(dodge_skill);
+        bool const is_trivial   = dodge_skill > att_rad_dodge_diff;
+
+        if (!critial_fail && (is_trivial || dodge_skill > rng(0, att_rad_dodge_diff))) {
+            target->add_msg_player_or_npc(m_rad_dodge_u, m_rad_dodge_npc);
+        } else if (one_in(att_rad_mutate_chance)) {
+            foe->mutate();
+        } else {
+            foe->add_msg_if_player(m_bad, m_rad_irradiate);
+            foe->radiation += rng(att_rad_dose_min, att_rad_dose_max);
         }
+      } break;
+    case att_manhack : {
+        z->moves -= att_cost_manhack;
+
+        // if the player can see it
+        if (g->u.sees(*z)) {
+            add_msg(m_warning, m_manhack_seen, z->name().c_str());
+        }
+        
+        auto const where = empty_neighbors.first[get_random_index(empty_neighbor_count)];
+
+        monster manhack {GetMType("mon_manhack")};
+        manhack.spawn(where.x, where.y);
+        manhack.friendly = z->friendly; // TODO: Factions
+        g->add_zombie(manhack);
+      } break;
+    case att_acid_pool :
+        z->moves -= att_cost_acid;
+
+        // if the player can see it
+        if (g->u.sees(*z)) {
+            add_msg(m_warning, m_acid_seen, z->name().c_str());
+        }
+        
+        // fill empty tiles with acid
+        for (size_t i = 0; i < empty_neighbor_count; ++i) {
+            auto const& p = empty_neighbors.first[i];
+            g->m.add_field(p.x, p.y, fd_acid, att_acid_density);
+        }
+
         break;
-    case 5: // Flavor text
-        const char *flavtext = "";
-        switch (rng(1, 4)) {
-        case 1:
-            flavtext = _("The %s gesticulates wildly!");
-            break;
-        case 2:
-            flavtext = _("The %s coughs up a strange dust.");
-            break;
-        case 3:
-            flavtext = _("The %s moans softly.");
-            break;
-        case 4:
-            flavtext = _("The %s's skin crackles with electricity.");
-            z->moves -= 80;
+    case att_flavor : {
+        // if the player can't see it, forget about it
+        if (!g->u.sees(*z)) {
             break;
         }
-        if( seen ) {
-            add_msg( m_warning, flavtext, z->name().c_str() );
+
+        auto const i = get_random_index(m_flavor);
+        add_msg(m_warning, m_flavor[i], z->name().c_str());
```

Comment:
The flavor texts, on the other hand, I really like in this container format. Really easy to JSONify from there.